core/src/packet.rs

```diff
@@ -44,8 +51,17 @@ pub struct Packet {
 }
 
 impl Packet {
-    pub fn new(data: [u8; PACKET_DATA_SIZE], meta: Meta) -> Self {
-        Self { data, meta }
+    pub fn new(meta: Meta, data: &[u8]) -> Result<Self> {
+        let mut packet_data = [0; PACKET_DATA_SIZE];
+        if data.len() != meta.size {
+            Err(Error::PackableError(PackableError::InvalidData))
+        } else {
+            packet_data.copy_from_slice(&data[..meta.size]);
+            Ok(Self {
+                data: packet_data,
```

Comment:
this is a memcpy(), right?  maybe instead make a mut Packet self and copy_from_slice() into that instead?